Pages/ProfilePage.cs

```diff
@@ -0,0 +1,307 @@
+        }
+
+        public bool isAtProfilePage()
+        {
+            return IsElementPresent(TabProfileBy);
+        }
+
+        public string SelectRandomLevel(By by, string[] dropDownList)
+        {
+            // Make sure Add New Button is clicked and Language Level visible
+            // Select any Language Level
+            Random random = new Random();
+            int index = random.Next(1, dropDownList.Count());
+            SelectElement dropDown = new SelectElement(_driver.FindElement(by));
+            dropDown.SelectByValue(dropDownList[index]);
+            return dropDownList[index];
+        }
+
+        public Tuple<string, string> AddRandomLanguage()
+        {
+            // Click Add New button
+            _driver.FindElement(BtnAddNewLanguageBy).Click();
+            Thread.Sleep(1000);
+
+            string charSet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ ";
+            Random random = new Random();
+            int length = random.Next(3, 21); // 21 is exclusive, so it generates number from 3 to 20
+            // Generate the random string
+            string randomString = new string(Enumerable.Range(0, length)
+                                      .Select(_ => charSet[random.Next(charSet.Length)]).ToArray());
+            WaitUtil.WaitVisible(_driver, FieldLanguageBy).SendKeys(randomString);
+
+            string langLevel = SelectRandomLevel(ChooseLanguageBy, LangDrop);
+            _driver.FindElement(BtnAddLanguageBy).Click();
+            Thread.Sleep(1000);
+
+            return new Tuple<string, string> (randomString, langLevel);
+            // https://www.c-sharpcorner.com/UploadFile/9b86d4/how-to-return-multiple-values-from-a-function-in-C-Sharp/
+        }
+
+        public Tuple<string, string> AddRandomSkill()
+        {
+            // Click Add New button
+            _driver.FindElement(BtnAddNewSkillBy).Click();
+            Thread.Sleep(1000);
+
+            string charSet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ ";
+            Random random = new Random();
+            int length = random.Next(3, 21); // 21 is exclusive, so it generates number from 3 to 20
+            // Generate the random string
+            string randomString = new string(Enumerable.Range(0, length)
+                                      .Select(_ => charSet[random.Next(charSet.Length)]).ToArray());
+            WaitUtil.WaitVisible(_driver, FieldSkillBy).SendKeys(randomString);
+
+            string skillLevel = SelectRandomLevel(ChooseSkillBy, SkillDrop);
+            _driver.FindElement(BtnAddSkillBy).Click();
+            Thread.Sleep(1000);
+
+            return new Tuple<string, string>(randomString, skillLevel);
+            // https://www.c-sharpcorner.com/UploadFile/9b86d4/how-to-return-multiple-values-from-a-function-in-C-Sharp/
+        }
+
+        public bool IsLanguageOrSkillPresent(int tabIndex, string name, string level, int rowNum = 0) // 1: Languages, 2: Skills
```

Comment:
so rather then check if they are present your test should think about state 

example before adding language you need to make sure there a no existing languages as this could result in false positive test pass
you will need to think about this for all the scenarios

the state is say that you have a test that is adding a language english, the second time you run this test you should delete any existing language so that the state is reset, this will guarantee that your test does not produce false negatives and passes always
you can assume that test execution order is not always guaranteed
imagine if you run the tests  in parallel mode the all tests will run at the same time
so you need to also make sure you do not wipe out other tests data
hence why when you remove data the best thing to do is delete only data you added for the test to be correct
e.g test 1 added english and french as language test 2 added hindi then when you reset data you already know what language you are adding for each test
there for on reset of the two test you should delete english and french for test 1 and hindi for test 2

Reply:
please fix this

Reply:
Created an afterscenario and added in function to delete added language or skill so that the state will be the same after execution.